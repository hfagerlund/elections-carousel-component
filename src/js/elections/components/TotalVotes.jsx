'use strict';
import React from 'react';
import PropTypes from 'prop-types';

import css from './totalvotes.scss';

export default class TotalVotes extends React.Component {
  render() {
    const totalVotes = this.props.riding.results
      ? this.props.riding.results
          .map(results => results.votes)
          .reduce((a, b) => a + b, 0)
      : null;

    return (
      <p className={css.totalVotes}>
        Total votes for this riding:
        <strong> {totalVotes}</strong> votes
      </p>
    );
  }
}

TotalVotes.propTypes = {
  riding: PropTypes.object
};
