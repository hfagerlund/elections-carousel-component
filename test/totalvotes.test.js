import TotalVotes from '../src/js/elections/components/TotalVotes';
import React from 'react';

import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';


describe('TotalVotes component', () => {
  describe('rendering tests', () => {
    it('should render vote total for the riding', () => {
      const tree = shallow(<TotalVotes riding={{}} />);
      expect(tree).toMatchSnapshot();
    });
	
    it('should match its snapshot with riding data', () => {
      var ridings = {
			"id": 1,
			"name": "Annapolis",
			"num": 1,
			"pollsReported": 57,
			"pollsTotal": 57,
			"results": [
				{
					"name": "Henry Spurr",
					"partyId": 13,
					"votes": 834,
					"isElected": false,
					"partyCode": "NDP"
				},
				{
					"name": "Stephen McNeil",
					"partyId": 12,
					"votes": 7709,
					"isElected": true,
					"partyCode": "LIB"
				},
				{
					"name": "Ginny Hurlock",
					"partyId": 14,
					"votes": 1390,
					"isElected": false,
					"partyCode": "PC"
				},
				{
					"name": "Ron Neufeld",
					"partyId": 11,
					"votes": 227,
					"isElected": false,
					"partyCode": "GRN"
				}
			]
		};


      const tree = renderer.create(
        <TotalVotes riding={ridings} />
      );
      expect(tree).toMatchSnapshot();
    });
  });
});

