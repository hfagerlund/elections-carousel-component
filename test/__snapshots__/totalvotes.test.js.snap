// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalVotes component rendering tests should match its snapshot with riding data 1`] = `
<p
  className="totalVotes"
>
  Total votes for this riding:
  <strong>
     
    10160
  </strong>
   votes
</p>
`;

exports[`TotalVotes component rendering tests should render vote total for the riding 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TotalVotes
    riding={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Total votes for this riding:",
        <strong>
           
        </strong>,
        " votes",
      ],
      "className": "totalVotes",
    },
    "ref": null,
    "rendered": Array [
      "Total votes for this riding:",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            null,
          ],
        },
        "ref": null,
        "rendered": Array [
          " ",
          null,
        ],
        "type": "strong",
      },
      " votes",
    ],
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Total votes for this riding:",
          <strong>
             
          </strong>,
          " votes",
        ],
        "className": "totalVotes",
      },
      "ref": null,
      "rendered": Array [
        "Total votes for this riding:",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              null,
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            null,
          ],
          "type": "strong",
        },
        " votes",
      ],
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
